WikiWatershed/model-my-watershed
# Add custom layer control + overlay layer support

Adds a custom layer control to the map, and adds support for overlay layers.

This took a bit longer than I expected because I needed to do some refactoring of how we implemented layer support so that this could be implemented easily. 

See commit messages for details.

This doesn't implement the layer control exactly as it's represented in the visual designs, but I thought that getting that right could be done by UI/UX.

As designed:

![screen shot 2015-08-31 at 5 24 19 pm](https://cloud.githubusercontent.com/assets/1042475/9590829/27853a8c-5005-11e5-8262-e28c31fc5b1d.png)

As implemented:

![screen shot 2015-08-31 at 5 24 41 pm](https://cloud.githubusercontent.com/assets/1042475/9590833/3122129a-5005-11e5-8b23-9f68279e8a50.png)

**Testing instructions**:

- Using the layer control, switch the active base map.
- Activate the "Overlays" tab of the layer control, and turn on and off the overlays for which you've loaded the data.
- Select an AoI.
- Any active overlays should still be on, and the layer control should still be present.

Known issues:

- The control is hidden by the analyze bar that appears when moving back from the analyze page to the draw page after having previously selected an AoI. I started to fix this in this PR, but it's not as simple as it first appears. We may need design support. https://github.com/WikiWatershed/model-my-watershed/issues/742

- The overlay layers that are also used as boundary layers have the same style.  https://github.com/WikiWatershed/model-my-watershed/issues/741

- The boundary label is drawn above map controls. https://github.com/WikiWatershed/model-my-watershed/issues/743

Connects to #629
